File: taskdesk-app/src/app/services/task.service.ts
import { Task } from './../model/Task.model';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { TASK_DESK_API } from './taskdesk.api';

@Injectable()
export class TaskService {

  constructor(private http: HttpClient) { }

  createOrUpdate(task: Task){
    if (task.id != null && task.id != '' ) {
      return this.http.put(`${TASK_DESK_API}/api/task`, task);
    }
    else
    {
      task.id = null;
      return this.http.post(`${TASK_DESK_API}/api/task`, task);
    }
  }

  findAll(page: number, count: number) {
    return this.http.get(`${TASK_DESK_API}/api/task/${page}/${count}`);
  }

  findById(id: string) {
    return this.http.get(`${TASK_DESK_API}/api/task/${id}`);
  }

  delete(id: string) {
    return this.http.delete(`${TASK_DESK_API}/api/task/${id}`);
  }

  findByParams(page: number, count: number, task : Task ){
    task.codigoTask = task.codigoTask == null ? 0 : task.codigoTask;
    task.titulo = task.titulo == null ? "uninformed" : task.titulo;
    task.prioridade = task.prioridade == null ? "uninformed" : task.prioridade;
    return this.http.get(`${TASK_DESK_API}/api/task/${page}/${count}/${task.codigoTask}/${task.titulo}/${task.prioridade}`);
  }
}

File: TaskDesk/src/main/java/com/welbertsantos/taskdesk/api/service/TaskService.java
package com.welbertsantos.taskdesk.api.service;

import org.springframework.data.domain.Page;
import org.springframework.stereotype.Component;

import com.welbertsantos.taskdesk.api.entity.Task;

@Component
public interface TaskService {

	Task createOrUpdate(Task task);
	
	Task findById(String id);
	
	void delete(String id);
	
	Page<Task> listTask (int page, int count);
	
	Page<Task> findByParameters(int page, int count, String titulo, String prioridade);
	
	Page<Task> findByCodigoTask (int page, int count, Integer codigoTask);
	
	Iterable<Task> findAll();
	
	
}
